lib/mutant/ast/regexp/transformer.rb

```diff
@@ -0,0 +1,82 @@
+module Mutant
+  module AST
+    module Regexp
+      class Transformer
+        include AbstractType
+
+        REGISTRY = {} # rubocop:disable MutableConstant
```

Comment:
@backus FWIW, I was going to suggest extracting the registry into it's own data structure. Often when you have a local object with lots of methods related to adding/removing/changing it, that's a sign that you have a concept that can be extracted into something more self contained.